tiledb/api/src/filter.rs

```diff
@@ -1,345 +1,216 @@
-            FilterOption::SCALE_FLOAT_FACTOR,
-            &c_factor as *const std::ffi::c_double as *const std::ffi::c_void,
-        )
-    }
-
-    pub fn get_float_factor(&self, ctx: &Context) -> TileDBResult<f64> {
+    pub fn get_float_factor(&self) -> TileDBResult<f64> {
         let mut c_factor: std::ffi::c_double = 0.0;
         self.get_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_FACTOR,
+            ffi::FilterOption::SCALE_FLOAT_FACTOR,
             &mut c_factor as *mut std::ffi::c_double as *mut std::ffi::c_void,
         )
         .map(|_| c_factor as f64)
     }
 
-    pub fn set_float_offset(
-        &self,
-        ctx: &Context,
-        offset: f64,
-    ) -> TileDBResult<()> {
-        let c_offset = offset as std::ffi::c_double;
-        self.set_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_OFFSET,
-            &c_offset as *const std::ffi::c_double as *const std::ffi::c_void,
-        )
-    }
-
-    pub fn get_float_offset(&self, ctx: &Context) -> TileDBResult<f64> {
+    pub fn get_float_offset(&self) -> TileDBResult<f64> {
         let mut c_factor: std::ffi::c_double = 0.0;
         self.get_option(
-            ctx,
-            FilterOption::SCALE_FLOAT_OFFSET,
+            ffi::FilterOption::SCALE_FLOAT_OFFSET,
             &mut c_factor as *mut std::ffi::c_double as *mut std::ffi::c_void,
         )
         .map(|_| c_factor as f64)
     }
 
-    pub fn set_webp_quality(
-        &self,
-        ctx: &Context,
-        quality: f32,
-    ) -> TileDBResult<()> {
-        let c_quality = quality as std::ffi::c_float;
-        self.set_option(
-            ctx,
-            FilterOption::WEBP_QUALITY,
-            &c_quality as *const std::ffi::c_float as *const std::ffi::c_void,
-        )
-    }
-
-    pub fn get_webp_quality(&self, ctx: &Context) -> TileDBResult<f32> {
-        let mut c_factor: std::ffi::c_float = 0.0;
+    pub fn get_positive_delta_max_window(&self) -> TileDBResult<u32> {
+        let mut c_width: std::ffi::c_uint = 0;
         self.get_option(
-            ctx,
-            FilterOption::WEBP_QUALITY,
-            &mut c_factor as *mut std::ffi::c_float as *mut std::ffi::c_void,
-        )
-        .map(|_| c_factor as f32)
-    }
-
-    pub fn set_webp_input_format(
-        &self,
-        ctx: &Context,
-        format: WebPFilterInputFormat,
-    ) -> TileDBResult<()> {
-        let c_format = format as std::ffi::c_uchar;
-        self.set_option(
-            ctx,
-            FilterOption::WEBP_INPUT_FORMAT,
-            &c_format as *const std::ffi::c_uchar as *const std::ffi::c_void,
+            ffi::FilterOption::POSITIVE_DELTA_MAX_WINDOW,
+            &mut c_width as *mut std::ffi::c_uint as *mut std::ffi::c_void,
         )
+        .map(|_| c_width as u32)
     }
 
     pub fn get_webp_input_format(
         &self,
-        ctx: &Context,
-    ) -> TileDBResult<WebPFilterInputFormat> {
+    ) -> TileDBResult<ffi::WebPFilterInputFormat> {
```

Comment:
Hm, we probably want to avoid having `ffi::` in the public APIs.

Reply:
I definitely wasn't fond of that either, but kinda left it as a "We'll clean up the api vs sys symbols at some point". For the interim, I'll go add a public use to just import `WebPFilterInputFormat` into the `api/filter` namespace to remove the `ffi` tokens.